src/js/map/maps.js

```diff
@@ -40,10 +43,23 @@ export class MapControl extends Component {
         };
 
         this.registerEvents();
+        if (config.watermarkEnabled) {
+            this.addWatermark();
+        }
 
         this.choropleth = new Choropleth(this, this.layerCache, this.layerStyler, config.map.choropleth);
     };
 
+    addWatermark() {
+        let watermarkWrapper = document.createElement('div');
+        $(watermarkWrapper).addClass('watermark-wrapper');
+        $('.map-watermark-wrapper')
+            .append(watermarkWrapper);
+
+        let watermarkRoot = createRoot(watermarkWrapper);
+        watermarkRoot.render(<Watermark/>);
+    }
```

Comment:
This block of code is repeated in tray,js and menu.js, could we add a single function to call from each place?